tests/core/test_library.py

```diff
@@ -167,6 +170,16 @@ def test_lookup_uris_returns_empty_list_for_dummy3_track(self):
         self.assertFalse(self.library1.lookup.called)
         self.assertFalse(self.library2.lookup.called)
 
+    def test_lookup_adds_source(self):
```

Comment:
I don't think this test proves what it is saying, since it makes `library.lookup()` return a track that already has the source attribute set.